test/hotspot/jtreg/compiler/predicates/assertion/TestSplitIfCloningDown.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+/*
+ * @test
+ * @bug 8330004
+ * @summary Sanity test to exercise code to clone a Template Assertion Predicate down in Split If.
+ * @requires vm.compiler2.enabled
+ * @run main/othervm -Xcomp -XX:LoopMaxUnroll=0
+ *                   -XX:CompileCommand=compileonly,compiler.predicates.assertion.TestSplitIfCloningDown::*
+ *                   compiler.predicates.assertion.TestSplitIfCloningDown
+ */
+
+/*
+ * @test id=no-flags
+ * @bug 8330004
+ * @summary Sanity test to exercise code to clone a Template Assertion Predicate down in Split If.
+ * @run main compiler.predicates.assertion.TestSplitIfCloningDown
+ */
+
+package compiler.predicates.assertion;
+
+public class TestSplitIfCloningDown {
```

Comment:
Just a sanity test that exercises the refactored code. There was no assert or anything alike when running the test before this refactoring.